### ISSUE TITLE
GC Stress Test does not run on Linux/Ubuntu

### ISSUE THREAD

**Issue by ramarag** [dotnet/coreclr#4536]
GC stress tests are failing with : 
http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/checked_ubuntu_tst_prtest/1055/testReport/junit/GC.Stress.Framework/ReliabilityFramework/ReliabilityFramework/

**[4/22/16 7:50:21 PM 1] Started
[4/22/16 7:50:21 PM 1] Argument: testmix_gc.config
[4/22/16 7:50:21 PM 1] Argument: /maximumExecutionTime:-1
[4/22/16 7:50:21 PM 1] Running tests...
[4/22/16 7:50:21 PM 1] Getting configuration...
[4/22/16 7:50:21 PM 1] Exception while running tests: System.IO.FileLoadException: Could not load file or assembly 'System.Xml.ReaderWriter, Version=4.0.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040 (FUSION_E_REF_DEF_MISMATCH))
File name: 'System.Xml.ReaderWriter, Version=4.0.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at ReliabilityConfig.GetTestsToRun(String testConfig)
   at ReliabilityConfig..ctor(String testConfig)
   at ReliabilityFramework.RunReliabilityTests(String testConfig, Boolean doReplay)
   at ReliabilityFramework.Main(String[] args)
[4/22/16 7:50:21 PM 1] Reliability framework is shutting down...
[4/22/16 7:50:21 PM 1] Shutdown w/ ret val of  -1**

**Comment by mjsabby** [comment]
I've seen FUSION_E_REF_DEF_MISMATCH with R2R image being tried to load. If someone investigates this and it does turn out to be that could we explain it here?

**Comment by swgillespie** [comment]
The original issue does not reproduce anymore, but the reliability framework still doesn't run on non-Windows platforms due to p/invokes into `kernel32.dll`. 

```
Unhandled Exception: System.DllNotFoundException: Unable to load DLL 'kernel32.dll': The specified module could not be found.
 (Exception from HRESULT: 0x8007007E)
   at ReliabilityFramework.IsDebuggerPresent()
   at ReliabilityFramework.MyDebugBreak(String extraData)
   at RFLogging.LogWorker()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
```

Shouldn't be too hard to fix.

**Comment by swgillespie** [comment]
Fixed by #11029
